### ISSUE TITLE
My system can't detect XIM

### ISSUE THREAD

**Issue by adenosie** [Riey/kime#111]
kime-xim doesn't work in my system so I ran a test program (just a simple modification from [here](https://gist.github.com/Determinant/19bbecb6db35312861f6cf9f54fdd3a5)) to check what's wrong:

```c
#include <X11/Xlib.h>
#include <X11/Xutil.h>
#include <X11/Xos.h>
#include <stdlib.h>
#include <stdio.h>
#include <locale.h>
#include <assert.h>

Display *dpy;
Window win;
int scr;

void send_spot(XIC ic, XPoint nspot) {
    XVaNestedList preedit_attr;
    preedit_attr = XVaCreateNestedList(0, XNSpotLocation, &nspot, NULL);
    XSetICValues(ic, XNPreeditAttributes, preedit_attr, NULL);
    XFree(preedit_attr);
}

int main() {
    /* fallback to LC_CTYPE in env */
    setlocale(LC_CTYPE, "");
    /* implementation-dependent behavior, on my machine it defaults to
     * XMODIFIERS in env */
    XSetLocaleModifiers("");

    /* setting up a simple window */
    dpy = XOpenDisplay(NULL);
    scr = DefaultScreen(dpy);
    win = XCreateSimpleWindow(dpy,
            XDefaultRootWindow(dpy),
            0, 0, 100, 100, 5,
            BlackPixel(dpy, scr),
            BlackPixel(dpy, scr));
    XMapWindow(dpy, win);

    /* initialize IM and IC */
    XIM xim = XOpenIM(dpy, NULL, NULL, NULL);
    if(!xim)
    {
        printf("no input method can be operated\n");
        return 1;
    }

    XIC ic = XCreateIC(xim,
                        /* the following are in attr, val format, terminated by NULL */
                        XNInputStyle, XIMPreeditNothing | XIMStatusNothing,
                        XNClientWindow, win,
                        NULL);
    if(!ic)
    {
        printf("can't create input context\n");
        return 1;
    }

    /* focus on the only IC */
    XSetICFocus(ic);
    /* capture the input */
    XSelectInput(dpy, win, KeyPressMask);
    XPoint spot;
    spot.x = 0;
    spot.y = 0;
    send_spot(ic, spot);

    static char *buff;
    size_t buff_size = 16;
    buff = (char *)malloc(buff_size);
    for (;;)
    {
        KeySym ksym;
        Status status;
        XEvent ev;
        XNextEvent(dpy, &ev);
        if (XFilterEvent(&ev, None))
            continue;
        if (ev.type == KeyPress)
        {
            size_t c = Xutf8LookupString(ic, &ev.xkey,
                                        buff, buff_size - 1,
                                        &ksym, &status);
            if (status == XBufferOverflow)
            {
                printf("reallocate to the size of: %lu\n", c + 1);
                buff = realloc(buff, c + 1);
                c = Xutf8LookupString(ic, &ev.xkey,
                                    buff, c,
                                    &ksym, &status);
            }
            if (c)
            {
                spot.x += 20;
                spot.y += 20;
                send_spot(ic, spot);
                buff[c] = 0;
                printf("delievered string: %s\n", buff);
            }
        }
    }
}

```

I made sure that kime-xim is running in background, and ran this program:
```sh
$ gcc test.c -lX11
$ kime-xim &
$ XMODIFIERS=@im=kime ./a.out
> no input method can be operated
```

If I use uim instead of kime-xim and replace XMODIFIERS accordingly, that works fine. so it must be kime or xim-rs that is in trouble.

**Comment by Riey** [comment]
Can you turn on log option with `kime-xim --log`?

**Comment by adenosie** [comment]
prints nothing... do I have to open some log file?

**Comment by Riey** [comment]
weird.. is kime-xim keep running state?

**Comment by adenosie** [comment]
yes. I tried with kime-xim in daemon, in background, in foreground. but there was no case kime-xim crashed.

**Comment by Riey** [comment]
Please post `xprop -root`

**Comment by adenosie** [comment]
```
GDK_VISUALS(INTEGER) = 1419, 1921
_NET_ACTIVE_WINDOW(WINDOW): window id # 0xe00009
ESETROOT_PMAP_ID(PIXMAP): pixmap id # 0xa00001
_XROOTPMAP_ID(PIXMAP): pixmap id # 0xa00001
I3_CONFIG_PATH(UTF8_STRING) = "/home/user_name/.config/i3/config"
I3_PID(CARDINAL) = 6739
I3_SOCKET_PATH(UTF8_STRING) = "/run/user/1000/i3/ipc-socket.6739"
_NET_CLIENT_LIST(WINDOW): window id # 0xe00009, 0x1000003, 0x400009
_NET_CLIENT_LIST_STACKING(WINDOW): window id # 0x1000003, 0xe00009, 0x400009
_NET_DESKTOP_NAMES(UTF8_STRING) = "1", "2"
_NET_DESKTOP_VIEWPORT(CARDINAL) = 0, 0, 0, 0
_NET_NUMBER_OF_DESKTOPS(CARDINAL) = 2
_NET_CURRENT_DESKTOP(CARDINAL) = 0
_NET_SUPPORTED(ATOM) = _NET_SUPPORTED, _NET_SUPPORTING_WM_CHECK, _NET_WM_NAME, _NET_WM_VISIBLE_NAME, _NET_WM_MOVERESIZE, _NET_WM_STATE_STICKY, _NET_WM_STATE_FULLSCREEN, _NET_WM_STATE_DEMANDS_ATTENTION, _NET_WM_STATE_MODAL, _NET_WM_STATE_HIDDEN, _NET_WM_STATE_FOCUSED, _NET_WM_STATE, _NET_WM_WINDOW_TYPE, _NET_WM_WINDOW_TYPE_NORMAL, _NET_WM_WINDOW_TYPE_DOCK, _NET_WM_WINDOW_TYPE_DIALOG, _NET_WM_WINDOW_TYPE_UTILITY, _NET_WM_WINDOW_TYPE_TOOLBAR, _NET_WM_WINDOW_TYPE_SPLASH, _NET_WM_WINDOW_TYPE_MENU, _NET_WM_WINDOW_TYPE_DROPDOWN_MENU, _NET_WM_WINDOW_TYPE_POPUP_MENU, _NET_WM_WINDOW_TYPE_TOOLTIP, _NET_WM_WINDOW_TYPE_NOTIFICATION, _NET_WM_DESKTOP, _NET_WM_STRUT_PARTIAL, _NET_CLIENT_LIST, _NET_CLIENT_LIST_STACKING, _NET_CURRENT_DESKTOP, _NET_NUMBER_OF_DESKTOPS, _NET_DESKTOP_NAMES, _NET_DESKTOP_VIEWPORT, _NET_ACTIVE_WINDOW, _NET_CLOSE_WINDOW, _NET_MOVERESIZE_WINDOW
_NET_WM_NAME(UTF8_STRING) = "i3"
_NET_SUPPORTING_WM_CHECK(WINDOW): window id # 0x600063
XIM_SERVERS(ATOM) = @server=uim, @server=kime
RESOURCE_MANAGER(STRING) = "*background:\t#32302f\n*color0:\t#282828\n*color1:\t#cc241d\n*color10:\t#b8bb26\n*color11:\t#fabd2f\n*color12:\t#83a598\n*color13:\t#d3869b\n*color14:\t#8ec07c\n*color15:\t#ebdbb2\n*color2:\t#98971a\n*color3:\t#d79921\n*color4:\t#458588\n*color5:\t#b16286\n*color6:\t#689d6a\n*color7:\t#a89984\n*color8:\t#928374\n*color9:\t#fb4934\n*foreground:\t#ebdbb2\nURxvt.font:\txft:Source Code Pro:pixelsize=16, xft:Noto Sans Mono CJK KR\nURxvt.pointerBlank:\ttrue\nURxvt.scrollBar:\tfalse\nXft:\tlcdfilter: lcddefault\nXft.antialias:\ttrue\nXft.autohint:\tfalse\nXft.hinting:\ttrue\nXft.hintstyle:\thintslight\nXft.rgba:\trgb\n"
_XKB_RULES_NAMES(STRING) = "evdev", "pc105", "us", "", ""
XFree86_has_VT(INTEGER) = 1
XFree86_VT(INTEGER) = 1

```

**Comment by Riey** [comment]
Just check, your version is 0.8.0 right? and `kime-xim --version` work?

**Comment by adenosie** [comment]
yes. I updated before turning X on

**Comment by Riey** [comment]
I can't reproduce this so I don't know what's wrong

**Comment by Riey** [comment]
It should be fixed in #178 can you check it?

**Comment by adenosie** [comment]
yes, it's fixed
